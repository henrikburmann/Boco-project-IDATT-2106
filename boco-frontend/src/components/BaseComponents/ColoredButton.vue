<template>
  <!-- Button with custom text and color -->
  <button
    class="flex items-center px-2 py-2 font-medium tracking-wide capitalize text-white transition-colors duration-200 transform rounded-md focus:outline-none focus:ring focus:ring-opacity-80 min-w-{20px}"
    :class="color"
  >
    {{ text }}
  </button>
</template>

<script>
/**
 * Colored button component with customizable text and 3 possible colors.
 * Colors:
 * blue (default),
 * red,
 * green
 */
export default {
  name: "ColoredButton",
  props: {
    text: String,
    buttonColor: {
      type: String,
      default: "blue",
    },
  },
  computed: {
    /**
     * Formats the tailwind css tags based on the color passed from parent
     */
    color() {
      if (this.buttonColor === "red") {
        return "bg-error-medium hover:bg-error-dark focus:ring-error-light";
      }
      if (this.buttonColor === "green") {
        return "bg-success-medium hover:bg-success-dark focus:ring-success-light";
      }
      return "bg-primary-medium hover:bg-primary-dark focus:ring-primary-light";
    },
  },
};
</script>
