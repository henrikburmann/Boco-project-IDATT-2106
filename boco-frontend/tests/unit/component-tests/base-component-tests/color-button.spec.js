import { mount } from "@vue/test-utils";
import ColoredButton from "@/components/BaseComponents/ColoredButton.vue";

describe("ColoredButtonComponent", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(ColoredButton, {
      //passing prop to component
      props: {
        text: "hei",
      },
    });
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it("is instantiated", () => {
    expect(wrapper.exists()).toBeTruthy();
  });
});
