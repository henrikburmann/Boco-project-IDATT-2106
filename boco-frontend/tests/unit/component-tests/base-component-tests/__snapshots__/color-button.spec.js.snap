// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColoredButtonComponent renders correctly 1`] = `
<div
  data-v-app=""
>
  
  <!-- Button with custom text and color -->
  <button
    class="flex items-center px-2 py-2 font-medium tracking-wide capitalize text-white transition-colors duration-200 transform rounded-md focus:outline-none focus:ring focus:ring-opacity-80 min-w-{20px} bg-primary-medium hover:bg-primary-dark focus:ring-primary-light"
  >
    hei
  </button>
  
</div>
`;
